sdk/identity/azure-identity/src/main/java/com/azure/identity/VisualStudioCodeCredential.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.implementation.IdentityClient;
+import com.azure.identity.implementation.IdentityClientBuilder;
+import com.azure.identity.implementation.IdentityClientOptions;
+import com.azure.identity.implementation.MsalToken;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Enables authentication to Azure Active Directory using data from Visual Studio Code
+ */
+public class VisualStudioCodeCredential implements TokenCredential {
```

Comment:
This type shouldn't be public. For now we are adding these credential types to the `DefaultAzureCredential` authentication flow but not exposing the credential types directly.

Reply:
Yeah, made it private.